test: Write integration tests for some releases commands using trycmd

kamilogorek opened the pull request:
This PR will act as the base for future integration testing.

AbhiPrasad reviewed (commented):
Looks good, I like the snapshots! The fact that they are under `tests/cmd` instead of `tests/commands` is kind of confusing though. Maybe we put them under `tests/fixtures/commands`?

I'll comment for now, but feel free to ping anytime will slam the approval.

kamilogorek commented on tests/common/mod.rs:
```diff
@@ -1,11 +1,59 @@
+use mockito::{mock, server_url, Matcher, Mock};
+use trycmd::TestCases;
+
+pub const UTC_DATE_FORMAT: &str = r#"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{6}Z"#;
```
I was thinking about this, but I want to reuse it across different regexes. We don't know what key will hold the timestamp in front.

kamilogorek commented on tests/common/mod.rs:
```diff
@@ -1,11 +1,59 @@
+pub const UTC_DATE_FORMAT: &str = r#"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{6}Z"#;
+
+pub fn create_testcase() -> TestCases {
+    let test_case = TestCases::new();
+    test_case
+        .env("SENTRY_URL", server_url())
```
`.env` is returning `&TestCases` and we need to return owned data here.

kamilogorek commented on tests/cmd/releases/releases-finalize-hyphen.trycmd:
````diff
@@ -0,0 +1,6 @@
+```
+$ sentry-cli releases finalize -wat-release
+? success
+Finalized release wat-release.
````
Its unfortunately based on the server response for now, and I didn't want to create 2 API mocks :3

kamilogorek commented on tests/cmd/releases/releases-finalize-hyphen.trycmd:
````diff
@@ -0,0 +1,6 @@
+```
+$ sentry-cli releases finalize -wat-release
````
Ok

kamilogorek commented:
Updated:

- changed structure to:
```
tests/integration
 - <command>
 - _api/<command>
 - _cases/<command>
```
- removed `common` in favor of putting all helpers inside `integration/mod.rs`
- renamed `create_testcase` to `register_test` and make it format path to the test itself (no more `tests/integration/_cases/` needed)
- made `with_response_file` format path to the response itself (no more `tests/integration/_api/` needed)
